src/beaker/sqlwarehouseutils.py

```diff
@@ -50,22 +54,26 @@ def _get_connection(self):
             schema=self.schema,
             session_configuration={"use_cached_result": results_caching},
         )
+        logging.info(f"Created new connection: {connection}")
         return connection
 
+    def close_connection(self):
+        try:
+            if self.connection:
+                self.connection.close()
+        except:
+            pass
+
     def execute_query(self, query_str):
-        connection = self._get_connection()
-        cursor = connection.cursor()
+        cursor = self.connection.cursor()
```

Comment:
Can we use context manager to avoid needing to close the cursor?

Reply:
Done